tests/helper.py

```diff
@@ -0,0 +1,145 @@
+
+# PropertyList is optional
+try:
+    import lsst.daf.base as dafBase
+except ImportError:
+    dafBase = None
+
+
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+
+
+def pl_constructor(loader, node):
+    """Construct an OrderedDict from a YAML file containing a PropertyList."""
+    pl = OrderedDict()
+    yield pl
+    state = loader.construct_sequence(node, deep=True)
+    for key, dtype, value, comment in state:
+        if dtype == "Double":
+            pl[key] = float(value)
+        elif dtype == "Int":
+            pl[key] = int(value)
+        elif dtype == "Bool":
+            pl[key] = True if value == "true" else False
+        else:
+            pl[key] = value
+
+
+if dafBase is None:
+    yaml.add_constructor("lsst.daf.base.PropertyList", pl_constructor)
+
+
+def readTestFile(filename):
+    """Read the named test file relative to the location of this helper
+
+    Parameters
+    ----------
+    filename : `str`
+        Name of file in the data directory.
+
+    Returns
+    -------
+    header : `dict`-like
+        Header read from file.
+    """
+    with open(os.path.join(TESTDIR, "data", filename)) as fd:
+        header = yaml.load(fd)
+    return header
+
+
+class UsefulAsserts:
+    """Class with helpful asserts"""
+
+    def assertCoordinatesConsistent(self, obsinfo, max_sep=1.0, amdelta=0.01):
+        """Check that SkyCoord, AltAz, and airmass are self consistent.
+
+        Parameters
+        ----------
+        obsinfo : `ObservationInfo`
+            Object to check.
+        max_sep : `float`, optional
+            Maximum separation between AltAz derived from RA/Dec headers
+            and that found in the AltAz headers.
+        amdelta : `float`, optional
+            Max difference between header airmass and derived airmass.
+
+        Raises
+        ------
+        AssertionError
+            Inconsistencies found.
+        """
+        self.assertIsNotNone(obsinfo.tracking_radec)
+        self.assertIsNotNone(obsinfo.altaz_begin)
+
+        # Is airmass from header close to airmass from AltAz headers?
+        # In most cases there is uncertainty over whether the elevation
+        # and airmass in the header are for the start, end, or middle
+        # of the observation.  Sometimes the AltAz is from the start
+        # but the airmass is from the middle so accuracy is not certain.
+        self.assertAlmostEqual(obsinfo.altaz_begin.secz.to_value(), obsinfo.boresight_airmass, delta=amdelta)
+
+        # Is AltAz from headers close to AltAz from RA/Dec headers?
+        sep = obsinfo.altaz_begin.separation(obsinfo.tracking_radec.altaz)
+        self.assertLess(sep.to_value(unit="arcmin"), max_sep)
+
+    def assertObservationInfo(self, file, checkWcs=True, wcsParams=None, **kwargs):
+        """Check contents of an ObservationInfo.
+
+        Parameters
+        ----------
+        file : `str`
+            Path to YAML file representing the header.
```

Comment:
I think it's better to pass in the header itself, so you can use this method with headers that don't live in yaml files.

Reply:
Then I'd need another helper function that takes the filename and calls this function with the contents.  I haven't had a need for that yet but maybe I should add it since probably I am going to have to move all the translators into obs_ packages and these helper routines will move into lsst.obs.metadata.tests so they can be called by obs_ package tests.